src/Http/Http.Extensions/src/DefaultProblemDetailsWriter.cs

```diff
@@ -66,7 +71,22 @@ public ValueTask WriteAsync(ProblemDetailsContext context)
                         contentType: "application/problem+json"));
     }
 
+    // Additional values are specified on JsonSerializerContext to support some values for extensions.
+    // For example, the DeveloperExceptionMiddleware serializes its complex type to JsonElement, which problem details then needs to serialize.
     [JsonSerializable(typeof(ProblemDetails))]
+    [JsonSerializable(typeof(JsonElement))]
+    [JsonSerializable(typeof(string))]
+    [JsonSerializable(typeof(decimal))]
+    [JsonSerializable(typeof(float))]
+    [JsonSerializable(typeof(double))]
+    [JsonSerializable(typeof(int))]
+    [JsonSerializable(typeof(long))]
+    [JsonSerializable(typeof(Guid))]
+    [JsonSerializable(typeof(Uri))]
+    [JsonSerializable(typeof(TimeSpan))]
+    [JsonSerializable(typeof(DateTime))]
+    [JsonSerializable(typeof(DateTimeOffset))]
```

Comment:
We should undo this.

Reply:
This is useful because it allows common type values as extension values to just work, e.g. adding a string value to problem details extensions.

Other than a small size cost when problem details aren't trimmed away, this doesn't hurt. Are you sure you want it removed?

Reply:
I don't mind keeping it for now and removing it with @brunolins16's JSON changes, but I don't think this is generally useful.

Reply:
Ok! I'd like to keep as is in this PR (rather than reverting to what it was, which doesn't work), and we can iterate on it during .NET 8 as we learn more.